x/gov/handler.go

```diff
@@ -0,0 +1,76 @@
+package gov
+
+import (
+	"reflect"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// Handle all "gov" type messages.
+func NewHandler(keeper Keeper) sdk.Handler {
+	return func(ctx sdk.Context, msg sdk.Msg) sdk.Result {
+		switch msg := msg.(type) {
+		case MsgDeposit:
+			return handleMsgDeposit(ctx, keeper, msg)
+		case MsgSubmitProposal:
+			return handleMsgSubmitProposal(ctx, keeper, msg)
+		case MsgVote:
+			return handleMsgVote(ctx, keeper, msg)
+		default:
+			errMsg := "Unrecognized gov Msg type: " + reflect.TypeOf(msg).Name()
+			return sdk.ErrUnknownRequest(errMsg).Result()
+		}
+	}
+}
+
+// Handle MsgSubmitProposal.
+func handleMsgSubmitProposal(ctx sdk.Context, keeper Keeper, msg MsgSubmitProposal) sdk.Result {
+
+	proposal := keeper.NewProposal(ctx, msg.Title, msg.Description, msg.ProposalType)
+
+	err := keeper.AddDeposit(ctx, proposal.ProposalID, msg.Proposer, msg.InitialDeposit)
+	if err != nil {
+		return err.Result()
+	}
+
+	proposalIDBytes, _ := keeper.cdc.MarshalBinaryBare(proposal.ProposalID)
```

Comment:
question: what's the purpose of using `Bare` here?

Reply:
ALSO let's not throw out any of these errors or use if it exists use `MustMarshalBinaryBare`

Reply:
Slightly more efficient - https://github.com/tendermint/go-amino/blob/549dbc3eb7eaaa507378f627e95ec54e6b793834/amino.go#L174 - should be constant length in this case presumably?

Reply:
Asked Jae, he said it doesn't matter in this case cause its a primitive type anyways.